src/SoftRobots/component/constraint/model/CableModel.inl

```diff
@@ -192,12 +192,11 @@ void CableModel<DataTypes>::init()
     {
         if (l_surfaceTopology.empty())
         {
-            msg_info() << "link to topology container of the surface should be set to ensure right behavior. First topology found in current context will be used.";
+            msg_info() << "Link to topology container of the surface should be set to ensure right behavior. First topology found in current context will be used.";
             l_surfaceTopology.set(this->getContext()->getMeshTopologyLink());
         }
-        m_topology = l_surfaceTopology.get();
 
-        if (m_topology == nullptr)
+        if (l_surfaceTopology.get() == nullptr)
             msg_error(this) << "There is no topology state associated with this node. "
                                 "To remove this error message, fix your scene possibly by "
                                 "adding a topology in this node";
```

Comment:
I recommend always adding the curly brackets even for a single if-instruction.

Also, if the topology is crucial for the component, you should set the component state to invalid:
```cpp
d_componentState.setValue(ComponentState::Invalid);
```
you can then use it later in the component to be sure the component has been initialized correctly. It usually avoid crashes